design/worker-config-drive.md

```diff
@@ -0,0 +1,178 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# worker-config-drive
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+## Summary
+
+In order to bring a host into the cluster as a worker we have to give
+it the correct ignition configuration as it boots. The contents of the
+Ignition file are available via the kubernetes API. Something needs to
+build a config drive image containing the ignition file and the
+baremetal operator needs to pass that formatted image to Ironic as
+part of provisioning the host.
+
+In order to avoid having the abstraction around Ironic break down,
+something that knows about the expectations of the image being
+deployed (especially the path where the Ignition file should be
+written within the config drive image) should build the drive image,
```

Comment:
s/drive image/configuration drive image/

This paragraph is really kind of confusing and actually caused me to misunderstand the entire meaning of what was attempting to be conveyed.

The abstraction is not about ironic as much as it is ignition specific behavior of how it has been designed to be invoked and the one or two places they will as a result load configuration from. Each class of image is in essence is hard wired to read configuration from a specific source.

Reply:
The point here is that we don't want the operator to build the config drive image because the operator doesn't know what's being provisioned. I'll try to find a better way to say that.